ui/src/tempVars/components/MeasurementsTemplateBuilder.tsx

```diff
@@ -0,0 +1,162 @@
+export default class MeasurementsTemplateBuilder extends PureComponent<
+  TemplateBuilderProps,
+  State
+> {
+  constructor(props) {
+    super(props)
+
+    const selectedDatabase = _.get(props, 'template.query.db', '')
+
+    this.state = {
+      databases: [],
+      databasesStatus: RemoteDataState.Loading,
+      selectedDatabase,
+      measurements: [],
+      measurementsStatus: RemoteDataState.Loading,
+    }
+  }
+
+  public async componentDidMount() {
+    await this.loadDatabases()
+    await this.loadMeasurements()
+  }
+
+  public render() {
+    const {
+      databases,
+      databasesStatus,
+      selectedDatabase,
+      measurements,
+      measurementsStatus,
+    } = this.state
+
+    return (
+      <div className="temp-builder measurements-temp-builder">
+        <div className="form-group">
+          <label>Meta Query</label>
+          <div className="temp-builder--mq-controls">
+            <div className="temp-builder--mq-text">SHOW MEASUREMENTS ON</div>
+            <DropdownPlaceholder rds={databasesStatus}>
+              <Dropdown
+                items={databases.map(text => ({text}))}
+                onChoose={this.handleChooseDatabaseDropdown}
+                selected={selectedDatabase}
+                buttonSize=""
+              />
+            </DropdownPlaceholder>
+          </div>
+        </div>
+        <TemplateMetaQueryPreview
+          items={measurements}
+          loadingState={measurementsStatus}
+        />
+      </div>
+    )
+  }
+
+  private async loadDatabases() {
+    const {source} = this.props
+
+    this.setState({databasesStatus: RemoteDataState.Loading})
+
+    try {
+      const {data} = await showDatabases(source.links.proxy)
+      const {databases} = parseShowDatabases(data)
+      const {selectedDatabase} = this.state
+
+      this.setState({
+        databases,
+        databasesStatus: RemoteDataState.Done,
+      })
+
+      if (!selectedDatabase) {
+        this.handleChooseDatabase(_.get(databases, 0, ''))
+      }
+    } catch (error) {
+      this.setState({databasesStatus: RemoteDataState.Error})
+      console.error(error)
+    }
+  }
+
+  private async loadMeasurements() {
+    const {template, source, onUpdateTemplate} = this.props
+    const {selectedDatabase} = this.state
+
+    this.setState({measurementsStatus: RemoteDataState.Loading})
+
+    try {
+      const {data} = await showMeasurements(
+        source.links.proxy,
+        selectedDatabase
+      )
+      const {measurementSets} = parseShowMeasurements(data)
+
+      const measurements = _.get(measurementSets, '0.measurements', [])
+
+      this.setState({
+        measurements,
+        measurementsStatus: RemoteDataState.Done,
+      })
+
+      const nextValues = measurements.map(value => {
+        return {
+          type: TemplateValueType.Measurement,
+          value,
+          selected: false,
+        }
+      })
+
+      if (nextValues[0]) {
+        nextValues[0].selected = true
```

Comment:
this also makes me wonder if it could be cool to go ahead and set state, and then call `handleChoose` on the first... but... meh, my idea is prob not as good of an approach actually.